ddtrace/internal/stats.py

```diff
@@ -0,0 +1,141 @@
+import collections
+import threading
+
+from ..utils.counter import Counter
+
+
+class Stats(object):
+    SPANS_STARTED = 'datadog.tracer.spans.started'
+    SPANS_STARTED_TYPE = 'increment'
+
+    SPANS_FINISHED = 'datadog.tracer.spans.finished'
+    SPANS_FINISHED_TYPE = 'increment'
+
+    ERROR_LOGS = 'datadog.tracer.log.errors'
+    ERROR_LOGS_TYPE = 'increment'
+
+    PATCH_ERROR = 'datadog.tracer.patch.error'
+    PATCH_ERROR_TYPE = 'increment'
+
+    PATCH_SUCCESS = 'datadog.tracer.patch.success'
+    PATCH_SUCCESS_TYPE = 'increment'
+
+    def __init__(self):
+        self._read_lock = threading.Lock()
+
+        self._last_values = collections.defaultdict(int)
+        self._counters = collections.defaultdict(Counter)
+        self._one_time_stats = set()
```

Comment:
I don't understand what this is used for? It looks like it's never read.

Reply:
this is one I haven't figured out yet.

this is used to indicate a metric whose counter should be removed from `self._counters` after it has been read/reset.

I did this because of the `datadog.tracer.patch.{error,success}` metrics, since they will only ever be incremented at start time, do we really need to report on them every flush?

maybe this is an example of a metric which should be reported as a `gauge` instead of an `increment`...

Reply:
I am just playing with these now in a sample app to get a feel for how we'll report on them.

Reply:
https://github.com/DataDog/dd-trace-py/blob/97be5a26cc31f3cc09cc34e269d30a776ebaea25/ddtrace/internal/stats.py#L117-L121

Reply:
So that sounds like it's needed for `gauge`, right? Values where you don't need to make any diff between the previous value and the current one?